src/components/SpacemeshApiClientMock/index.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from "react";
+import { ISpacemeshApi } from '../../model';
+import { Transaction, TransactionStatus } from 'src/types';
+
+interface IProps extends ISpacemeshApi{
+    apiReady: boolean;
+}
+
+interface IState {
+    connectedToApi: boolean;
+}
+
+const SpacemeshApiClientMock = <P extends IProps>(Component: React.ComponentType<P>) => {
+    return class SpacemeshApiClientMock extends React.Component<P & IProps, IState> implements ISpacemeshApi {
+        constructor(props: P & IProps) {
+            super(props);
+            this.state = {
+                connectedToApi: false,
+            }
+        }
+
+        /**
+         * Broadcast a transaction to the Spacemesh newwork.
+         */
+        public postTransaction = (tx: Transaction): Promise<boolean> => {
+            return new Promise(resolve => {
+                setTimeout(() => {
+                    resolve(true);
+                }, 2000);
+            });
+        }
+
+        /**
+         * Get SMC balance of address.
+         */
+        public getAddressBalance = (address: string): Promise<number> => {
+            return new Promise(resolve => {
+                setTimeout(() => {
+                    resolve(115);
+                }, 2000);
+            });
+        }
+
+        /**
+         * Get address from transaction.
+         */
+        public getAddressFromTransaction = (tx: Transaction): Promise<string> => {
+            return new Promise(resolve => {
+                setTimeout(() => {
+                    resolve(tx.from);
+                }, 2000);
+            });
```

Comment:
@avive I think this should be, fetch an address list of transactions. That is transactions that an address has made previously. This is useful when restoring a wallet.

Reply:
agree